psi4/src/psi4/libmints/bessel.cc

```diff
@@ -73,7 +74,7 @@ int BesselFunction::tabulate(const double accuracy) {
 	// where F_j(z) = e^(-z) * (z^2/2)^j / j!
 	int lmax = lMax + TAYLOR_CUT;
 	
-	double F[order + 1]; // F_j above
+	std::vector<double> F(order + 1); // F_j above
```

Comment:
I dont think we want to use a `std::vector` here. `F` is pretty small and probably needs to remain on the stack for performance reasons, is there a reason for this change? Paging @andysim @jturney.

Reply:
@raimis is right about this.  The order could possibly be gleaned from the LIBINT angular momentum macros.  I'll dig around and see what I can find.

Reply:
And by that, I mean the maximum value order could possibly take (it's related to the supported Ang. Mom.).  We could use that to allocate a chunk on the stack that's always sufficient.

Reply:
Ok, so there is a way to figure out the maximum size at compile time.  However, this allocation happens infrequently enough that I don't think it's a performance concern.  I vote to pull these changes in as they are and we can take an optimization pass through as part of the screening / gradients implementation.  There are a number of small allocations in the ECP code that should be cleaned up, as well as a couple of mismatched `new` / `free` calls.  I don't think it's a good idea to have massive scope creep and fix them all in this PR.

Reply:
@dgasmith @andysim how can we proceed with this issue?

Should it become something like:
```c++
double F[MAX_AM_ERI + 1];
```

Reply:
I found that it was hardcoded to 200 in https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libmints/ecpint.cc#L355 

I say leave it as you have it: `std::vector<double> ... `.  It's not allocated that frequently.